gdpr.php

```diff
@@ -255,8 +255,7 @@ function gdpr_civicrm_buildForm($formName, $form) {
     }
   }
   if ($formName == 'CRM_Contribute_Form_Contribution_ThankYou') {
-    $trxnId = $form->getVar('_trxnId');
-    $cid    = CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_Contribution', $trxnId, 'contact_id', 'trxn_id');
+    $cid = $form->_membershipContactID;
```

Comment:
I think better to use `$form->_contactID` instead of the above variable. @aniesshsethh Can you verify if this works for you?

Reply:
Yeah, that works too, the reason I didn't used that in order to handle the condition where somebody else is registering for somebody else, I may be wrong there though. Please confirm and I Will change.

Reply:
$form->_contactID works for us (just tested), but I agree wih @aniesshsethh this should be doublechecked. What if a person pays for someone else with a Pay Now link. The right person should be identified in those scenarios, not just the logged in user.